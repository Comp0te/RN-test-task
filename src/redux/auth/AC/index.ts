import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';

export enum ActionTypes {
  AUTH_LOGOUT = 'AUTH_LOGOUT',
  AUTH_SET_IS_AUTH_USER = 'AUTH_SET_IS_AUTH_USER',
}

export const Actions = {
  logout: () => createAction(ActionTypes.AUTH_LOGOUT),
  setIsAuthUser: (data: boolean) => createAction(ActionTypes.AUTH_SET_IS_AUTH_USER, {data}),
};

export type Actions = ActionsUnion<typeof Actions>;
