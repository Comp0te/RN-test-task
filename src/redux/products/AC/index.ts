import { createAction, ActionsUnion } from '../../../shared/helpers/createAction';
import { ProductModel } from '../../../shared/models/product.model';

export enum ActionTypes {
  SET_PRODUCTS_DATA = 'SET_PRODUCTS_DATA',
}

export const Actions = {
  setProductsData: (data: ProductModel[]) => createAction(ActionTypes.SET_PRODUCTS_DATA, {data}),
};

export type Actions = ActionsUnion<typeof Actions>;
