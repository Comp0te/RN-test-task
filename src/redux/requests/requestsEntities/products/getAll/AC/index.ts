import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ProductModel } from '../../../../../../shared/models/product.model';

export enum ActionTypes {
  PRODUCTS_GET_ALL_REQUEST = 'PRODUCTS_GET_ALL_REQUEST',
  PRODUCTS_GET_ALL_REQUEST_SUCCESS = 'PRODUCTS_GET_ALL_REQUEST_SUCCESS',
  PRODUCTS_GET_ALL_REQUEST_FAIL = 'PRODUCTS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllProducts: () => createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST),
  getAllProductsSuccess: (data: ProductModel[]) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllProductsFail: (errors: string) => {
    return createAction(ActionTypes.PRODUCTS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
