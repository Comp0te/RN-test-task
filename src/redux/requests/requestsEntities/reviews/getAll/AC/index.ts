import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { ReviewResponse } from '../../../../../../shared/models/review.model';

export enum ActionTypes {
  REVIEWS_GET_ALL_REQUEST = 'REVIEWS_GET_ALL_REQUEST',
  REVIEWS_GET_ALL_REQUEST_SUCCESS = 'REVIEWS_GET_ALL_REQUEST_SUCCESS',
  REVIEWS_GET_ALL_REQUEST_FAIL = 'REVIEWS_GET_ALL_REQUEST_FAIL',
}

export const Actions = {
  getAllReviews: (data: string) => createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST, {data}),
  getAllReviewsSuccess: (data: ReviewResponse[]) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_SUCCESS, {data});
  },
  getAllReviewsFail: (errors: string) => {
    return createAction(ActionTypes.REVIEWS_GET_ALL_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
