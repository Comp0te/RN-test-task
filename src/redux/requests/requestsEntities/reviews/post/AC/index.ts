import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { IReviewPostInput } from '../../../../../../shared/services/reviews.service';
import { ReviewResponse } from '../../../../../../shared/models/review.model';
import { AjaxError } from 'rxjs/ajax';

export enum ActionTypes {
  REVIEW_POST_REQUEST = 'REVIEW_POST_REQUEST',
  REVIEW_POST_REQUEST_SUCCESS = 'REVIEW_POST_REQUEST_SUCCESS',
  REVIEW_POST_REQUEST_FAIL = 'REVIEW_POST_REQUEST_FAIL',
}

export const Actions = {
  postReview: (data: IReviewPostInput) => createAction(ActionTypes.REVIEW_POST_REQUEST, {data}),
  postReviewSuccess: (data: ReviewResponse) => {
    return createAction(ActionTypes.REVIEW_POST_REQUEST_SUCCESS, {data});
  },
  postReviewFail: (errors: AjaxError) => {
    return createAction(ActionTypes.REVIEW_POST_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
