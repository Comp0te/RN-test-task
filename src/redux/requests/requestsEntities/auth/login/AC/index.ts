import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  LOGIN_REQUEST = 'LOGIN_REQUEST',
  LOGIN_REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS',
  LOGIN_REQUEST_FAIL = 'LOGIN_REQUEST_FAIL',
}

export const Actions = {
  login: (data: AuthInput) => createAction(ActionTypes.LOGIN_REQUEST, {data}),
  loginSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.LOGIN_REQUEST_SUCCESS, {data});
  },
  loginFail: (errors: string) => {
    return createAction(ActionTypes.LOGIN_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
