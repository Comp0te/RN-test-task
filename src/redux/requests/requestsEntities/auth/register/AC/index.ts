import { createAction, ActionsUnion } from '../../../../../../shared/helpers/createAction';
import { AuthInput } from '../../../../../../shared/services/auth.service';
import { AuthResponse } from '../../../../types';

export enum ActionTypes {
  REGISTER_REQUEST = 'REGISTER_REQUEST',
  REGISTER_REQUEST_SUCCESS = 'REGISTER_REQUEST_SUCCESS',
  REGISTER_REQUEST_FAIL = 'REGISTER_REQUEST_FAIL',
}

export const Actions = {
  register: (data: AuthInput) => createAction(ActionTypes.REGISTER_REQUEST, {data}),
  registerSuccess: (data: AuthResponse) => {
    return createAction(ActionTypes.REGISTER_REQUEST_SUCCESS, {data});
  },
  registerFail: (errors: string) => {
    return createAction(ActionTypes.REGISTER_REQUEST_FAIL, {errors});
  },
};

export type Actions = ActionsUnion<typeof Actions>;
