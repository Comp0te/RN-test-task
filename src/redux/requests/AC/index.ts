import * as login from '../requestsEntities/auth/login/AC';
import * as register from '../requestsEntities/auth/register/AC';
import * as getAllProducts from '../requestsEntities/products/getAll/AC';
import * as getAllReviews from '../requestsEntities/reviews/getAll/AC';
import * as postReview from '../requestsEntities/reviews/post/AC';

export const requestsAC = {
  login,
  register,
  getAllProducts,
  getAllReviews,
  postReview,
};