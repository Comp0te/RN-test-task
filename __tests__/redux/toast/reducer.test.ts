import { Actions, ActionTypes } from '../../../src/redux/toast/AC';
import {
  toastReducer,
  initialState as toastInitialState,
} from '../../../src/redux/toast/reducers';

describe('Redux toast reducers', () => {
  it(`update state after ${ActionTypes.TOAST_SHOW} action`, () => {
    expect(toastReducer(
      toastInitialState,
      Actions.showToast('message')),
    ).toMatchSnapshot();
  });

  it(`update state after ${ActionTypes.TOAST_RESET} action`, () => {
    expect(toastReducer(
      toastInitialState,
      Actions.hideToast()),
    ).toMatchSnapshot();
  });
});
