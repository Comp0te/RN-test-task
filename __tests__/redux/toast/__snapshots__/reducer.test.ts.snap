// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux toast reducers update state after TOAST_RESET action 1`] = `
Object {
  "message": null,
}
`;

exports[`Redux toast reducers update state after TOAST_SHOW action 1`] = `
Object {
  "message": "message",
}
`;
